src/abacus-ts/websocket/lib/reconnectingWebsocket.ts

```diff
@@ -24,111 +23,207 @@ export class ReconnectingWebSocket {
 
   private readonly backoffMultiplier: number;
 
+  private ws: WebSocketConnection | null = null;
+
+  private currentId: number = 0;
```

Comment:
does currentId ever get decremented?

Reply:
nope, only up

Reply:
ah a new session resets this?

Reply:
got it

Reply:
yeah it increments anytime we restart the websocket on purpose - either because of page visibility or network connectivity or the socket died and we are restarting it.